apps/worker/src/app/workflow/usecases/send-message/send-message-push.usecase.ts

```
@@ -222,27 +238,14 @@ export class SendMessagePush extends SendMessageBase {
     });
 
     if (!integration) {
-      await this.sendNoActiveIntegrationError(command.job);
+      await this.createExecutionDetailsError(DetailEnum.SUBSCRIBER_NO_ACTIVE_INTEGRATION, command.job);
 
-      return undefined;
+      throw new Error(DetailEnum.SUBSCRIBER_NO_ACTIVE_INTEGRATION);
```

Why are we throwing in this case instead of return `{ status: 'failed'. ...`?

---

Good question. This is because of the current logic we had, where we looped over the push channel integration and counted errors outside of this loop in case of errors. But this is a good point: We can make it more graceful.